ilastik/utility/data_url.py

```diff
@@ -0,0 +1,522 @@
+from abc import ABC, abstractmethod
+from typing import ClassVar, Dict, Mapping, Set, Tuple, TypeVar, Sequence, Optional, List, Type, Iterable, Union, cast
+from pathlib import PurePosixPath, Path
+import errno
+import glob
+import os
+import re
+from urllib.parse import urlencode, urlsplit, parse_qs
+import enum
+
+import numpy as np
+from typing_extensions import final
+import z5py
+import h5py
+
+from lazyflow.utility.pathHelpers import lsH5N5, globH5N5, globNpz
+
+# pyright: strict
+
+
+class Scheme(enum.Enum):
+    HTTP = "http"
+    HTTPS = "https"
+    FILE = "file"
+
+    @classmethod
+    def from_string(cls, value: str) -> "Scheme":
+        if value == "":
+            return Scheme.FILE
+        for item in cls:
+            if item.value == value:
+                return item
+        raise ValueError(f"Could not convert {value} to a valid Scheme")
+
+    @classmethod
+    def contains(cls, value: str) -> bool:
```

Comment:
This method seems to be used only once, and there it could also be done by constructing in a `try... except``` with raise from there... 

I think it's a bit weird that his enables:

```Python
>>> x = Scheme("http")
>>> x.contains("https")
True
```